llvm/test/CodeGen/X86/code-model-elf.ll

```diff
@@ -57,6 +57,7 @@ target triple = "x86_64--linux"
 @opaque = external dso_local global %t
 @forced_small_data = dso_local global [10 x i32] zeroinitializer, code_model "small", align 16
 @forced_large_data = dso_local global [10 x i32] zeroinitializer, code_model "large", align 16
+@__ehdr_start = external dso_local global i8
```

Comment:
Do users declare `__ehdr_start` as hidden in C source code? Is that why it is known to be DSO local? Otherwise, it's a declaration, so I would assume that you get the large access pattern by default.

Reply:
it can be declared with hidden visibility, e.g. https://github.com/llvm/llvm-project/blob/e6fdbd17769d626c66e087b0fd9bbda0ea69194d/compiler-rt/lib/profile/InstrProfilingPlatformLinux.c#L159

if the declaration type has a size, we respect the size of the type in regards to the large data threshold since we assume the definition has the same type

we have a similar issue with __start/__stop symbols declarations, will send out a patch for that soon